### ISSUE TITLE
./nodebb build should only build static assets

### ISSUE THREAD

**Issue by MozJon** [NodeBB/NodeBB#6426]
While setting up an environment using docker I've noticed something that seems unexpected.
`./nodebb build`: 

1. Checks for a `config.json` otherwise starts the web installer. (Which for obvious reasons doesn't work in a Docker build step.) 
2. Expects an existing MongoDB connection

- **NodeBB version: 1.7.5 (Needed at this version due to a plugin requirement for importing our old database** 
- **NodeBB git hash: 7df191412de9dfd88c3c9b919b6989fd18b5d445** 
- **Database type: Mongo**
- **Database version: 3.6.3** 

- **Exact steps to cause this issue:**
1. Setup a docker image with NodeJS and other misc requirements
2. Install NodeBB version 1.7.5
3. Install desired plugin packages via npm
4. From NodeBB application directory execute `./nodebb build`
- **What you expected:**
Static assets are built without any alterations to the DB
- **What happened instead:**
Lack of existing config launches the web installer.

**Comment by pitaj** [comment]
Without a config file and a database connection, the build process can't get a list of activated plugins. It needs to know what plugins are activated, otherwise it would build unnecessary files.

**Comment by MozJon** [comment]
I see. Alright then I misunderstood. Thanks for the info.

**Comment by MozJon** [comment]
I had been thinking about your reply and I was wondering a few things.

Why not allow for a `plugins.json` that tells NodeBB to build the assets for all of those, activated or not. This means you can build all installed assets without a DB connection. The activation should then be handled via the UI and simply be a flag to **show** and **execute** the plugin code.

**Comment by MozJon** [comment]
To expand on that I know you don't want to build unnecessary files but at the same time the responsibility should be on the developer not to bloat out their `plugins.json` This can be a separate command from `build` if you'd rather not alter the existing functionality.

**Comment by pitaj** [comment]
Here's the thing: almost every single command that the CLI offers requires a config.json and a database connection.

Crafting a separate command that does not is a lot of work because I'd have to change essentially the whole thing.

**Comment by funnylookinhat** [comment]
@pitaj Yeah - this leaves those of us trying to deploy w/ Docker images in a pretty tough spot.  Ideally, we would be able to run a command to generate those static assets in our build process before we push the image to the registry so that images can be static and start up quickly.

I wonder if anyone else could use that functionality; maybe it would warrant exploring building a one-off tool just to generate the static assets using some JSON file to specify plugins like @MozJon suggested.

**Comment by pitaj** [comment]
I'll accept a PR. I really know nothing about Docker, so I can't propose alternate solutions.

**Comment by pitaj** [comment]
> exploring building a one-off tool just to generate the static assets using some JSON file to specify plugins

I've looked into this somewhat and I think it could be done after these things happen:

- Change [this](https://github.com/NodeBB/NodeBB/blob/master/src/meta/css.js#L101) to use information loaded in `plugins.prepareForBuild` or something
- Bypass [this line](https://github.com/NodeBB/NodeBB/blob/master/src/meta/build.js#L192), instead doing `beforeBuild` things in your standalone utility

and of course there's more stuff to change to make all of that work

**Comment by funnylookinhat** [comment]
Great - thanks for pointing those out!

**Comment by coderafei** [comment]
Is there a better way?

**Comment by bukajsytlos** [comment]
any chance there was progress made here?
